src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -0,0 +1,179 @@
+                        property.Schema.Type = AllSchemaTypes.Uri;
+                }
+            }
+        }
+
+        public static void UpdateSubscriptionIdForAllResource(this CodeModel codeModel)
+        {
+            bool setSubParam = false;
+            foreach (var operationGroup in codeModel.OperationGroups)
+            {
+                foreach (var op in operationGroup.Operations)
+                {
+                    foreach (var p in op.Parameters)
+                    {
+                        //updater the first subscriptionId to be 'method'
+                        if (!setSubParam && p.Language.Default.Name.Equals("subscriptionId", StringComparison.OrdinalIgnoreCase))
+                        {
+                            setSubParam = true;
+                            p.Implementation = ImplementationLocation.Method;
+                        }
+                        //updater the first subscriptionId to be 'method'
+                        if (p.Language.Default.Name.Equals("apiVersion", StringComparison.OrdinalIgnoreCase))
+                        {
+                            p.Implementation = ImplementationLocation.Client;
+                        }
+                    }
+                }
+            }
+        }
+
+        private static Regex? _regex;
+
+        internal static void ApplyRenameRules(IReadOnlyDictionary<string, string> renameRules)
+        {
+            foreach ((var key, var value) in renameRules)
+            {
+                RenameRules.Add(key, value);
+            }
+            BuildRegex();
+        }
+
+        private static void BuildRegex()
+        {
+            if (RenameRules.Count == 0)
+                return;
+            var regexRawString = string.Join("|", RenameRules.Keys);
+            // we are building the regex that matches
+            // 1. it starts with a lower case letter or a digit which is considered as the end of its previous word
+            //    or it is at the beginning of this string
+            //    or we hit a word separator including \W, _, ., @, -, spaces
+            // 2. it either should match one of the candidates in the rename rules dictionary
+            // 3. it should followed by a upper case letter which is considered as the beginning of next word
+            //    or it is the end of this string
+            //    or we hit a word separator including \W, _, ., @, -, spaces
+            _regex = new Regex(@$"([\W|_|\.|@|-|\s|\$\da-z]|^)({regexRawString})([\W|_|\.|@|-|\s|\$A-Z]|$)");
+        }
+
+        public static void UpdateAcronyms(this IEnumerable<Schema> allSchemas, IReadOnlyDictionary<string, string> renameRules)
+        {
+            ApplyRenameRules(renameRules);
+            foreach (var schema in allSchemas)
+            {
+                TransformSchema(schema);
+            }
+        }
+
+        public static IEnumerable<Schema> GetAllSchemas(this CodeModel codeModel)
```

Comment:
Id prefer if we did this inline in the library ctor since this opens the opportunity to get called multiple times.

Reply:
makes sense.
Caching this here might cause concurrency issues while running test cases.
I removed this function and put it inline in outputLibrary